articles/building-a-basic-api-with-nodejs/index.md

```diff
@@ -0,0 +1,222 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/building-a-basic-api-with-nodejs/
+title: How To Build and Test a Node.js REST API with Express on Ubuntu 18.04
+description: In this guide, you will build a Node.js REST API with the Express Framework, expose it to the internet with Ngrok.
+author: adrian-murage
+date: 2020-08-25T00:00:00-07:00
+topics: [Node.js]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/building-a-basic-api-with-nodejs/hero.jpg
+    alt: basic API with Node.js image example
+---
+In this guide, you will build a [Node.js](https://nodejs.dev/learn) REST API with the [Express Framework](http://expressjs.com/), expose it to the internet with [Ngrok](https://ngrok.com/docs) and make test requests to it on [Postman](https://www.postman.com/).
+<!--more-->
+
+### Introduction
+Web services based on REST Architecture are known as RESTful web services. These web services use HTTP methods to implement the concept of REST architecture. A RESTful web service usually defines a [URI](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier) which provides resource representation such as JSON and set of HTTP Methods.
+
+### Prerequisites
+Before you begin this guide you'll need the following:
+
+- A non-root user with sudo privileges on Ubuntu 18.04
+- [Node.js](https://github.com/nodesource/distributions/blob/master/README.md#debinstall) installed.
+- [Npm](https://github.com/nodesource/distributions/blob/master/README.md#debinstall) installed.
+- [Postman](https://www.postman.com/product/api-client/) installed.
+- [Ngrok](https://ngrok.com/download) executable downloaded.
+
+### Step 1 — Build and Run an Express Server with Node.js
+In this step, you will write up an Express server with Node.js and run it locally.
+
+First, Setup [Express](http://expressjs.com/en/starter/installing.html).
+After the setup you should have one directory ```node_modules``` and two files ```package.json``` & ```package-lock.json```.
```

Comment:
Sorry, I may have been slightly unclear regarding the proper markdown formatting for code. Single line code should only by surrounded by one backtick which you had before whereas multi-line code blocks (where you add js or json etc. for syntax highlighting) should be surrounded by three. https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#code

Reply:
Thanks. I'll revert those changes.